src/macaron/policy_engine/prelude/policy.dl

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved. */
+/* Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/. */
+
+/**
+ *  Relations for creating policies about repositories, and controlling which repositories must satisfy their
+ *  requirements.
+ */
+
+/*
+ * Restrict the policy evaluation to a specific set of analyses.
+ */
+.decl restrict_to_analysis(analysis_id: number)
+.decl apply_to_analysis(analysis_id: number)
+apply_to_analysis(analysis) :-
+    analysis(analysis, _, _, _),
+    restrict_to_analysis(analysis).
+apply_to_analysis(analysis) :-
+    analysis(analysis, _, _, _),
+    !restrict_to_analysis(_).
+
+
+/**
+ *   A policy that specifies a set of rules about repositories.
+ *
+ *      policy_id: The unique identifier for this policy.
+ *      repo: the id field of a repository relation.
+ */
+.decl Policy(policy_id: symbol, repo: number, message: symbol)
+
+/**
+ *  Specifies that a repository repo must satisfy the policy policy_id: that the fact Policy(policy_id, repo)
+ *  must exist.
+ */
+.decl apply_policy_to(policy_id: symbol, repo: number)
```

Comment:
In the documentation, it is mentioned that `that the fact Policy(policy_id, repo)  must exists`. However, I don't see any rules defined for the relation between apply_policy_to and Policy, what would be the best way to understand this rule?

Reply:
These are the interface for defining policies, as used in the example in `tests/policy_engine/resources/policies/test_policy.dl`